examples/memcached.md

```diff
@@ -0,0 +1,369 @@
+
+In this mode, the PMEM-CSI driver is referenced through its name
+(usually `pmem-csi.intel.com`). No storage classes are needed. That
+name and other parameters in the deployment can be modified with
+[`kustomize`](https://github.com/kubernetes-sigs/kustomize). Here's
+how one can change the namespace, volume size or add additional
+command line parameters:
+
+```sh
+mkdir -p my-memcached-deployment
+
+cat >my-memcached-deployment/kustomization.yaml <<EOF
+namespace: demo
+bases:
+ - github.com/intel/pmem-csi/deploy/kustomize/memcached/ephemeral
+patchesStrategicMerge:
+- update-deployment.yaml
+EOF
+
+cat >my-memcached-deployment/update-deployment.yaml <<EOF
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: pmem-memcached
+spec:
+  template:
+    spec:
+      containers:
+      - name: memcached
+        env:
+        - name: MEMCACHED_ARGS
+          value: --conn-limit=500
+      volumes:
+      - name: data-volume
+        csi:
+          volumeAttributes:
+            size: 500Mi
+EOF
+```
+
+These commands then create the namespace and the deployment:
+```sh
+kubectl create ns demo
+kubectl create --kustomize my-memcached-deployment
+```
+
+The expected output is:
+```
+namespace/demo created
+service/pmem-memcached created
+deployment.apps/pmem-memcached created
+```
+
+Eventually, there will be one pod with memcached running:
+```sh
+kubectl wait --for=condition=Ready -n demo pods -l app.kubernetes.io/name=pmem-memcached
+```
+
+```
+pod/pmem-memcached-96f6b5986-hdx6n condition met
+```
+
+How to access a service from remote varies between cluster
+installations and how they handle incoming connections. Therefore we
+use [`kubectl
+port-forward`](https://kubernetes.io/docs/tasks/access-application-cluster/port-forward-access-application-cluster/)
+to access the service. Run this command in one shell and keep it
+running there:
+```sh
+kubectl port-forward -n demo service/pmem-memcached 11211
+```
+
+```
+Forwarding from 127.0.0.1:11211 -> 11211
+Forwarding from [::1]:11211 -> 11211
+```
+
+In another shell we can now use `telnet` to connect to memcached:
+```sh
+telnet localhost 11211
+```
+
+```
+Trying ::1...
+Connected to localhost.
+Escape character is '^]'.
+```
+
+Memcached accepts simple plain-text commands. To set a key with 10
+characters, a time-to-live of 500 seconds and default flags, enter:
+```
+set demo_key 0 500000 10
+I am PMEM.
+```
+
+Memcached acknowledges this with:
+```
+STORED
+```
+
+We can verify that the key exists with:
+```
+get demo_key
+```
+
+```
+VALUE demo_key 0 10
+I am PMEM.
+END
+```
+
+To disconnect, use:
+```
+quit
+```
+
+```
+Connection closed by foreign host.
+```
+
+The following command verifies the data was stored in a persistent
+memory data volume:
+```sh
+pod=$(kubectl get -n demo pods -l app.kubernetes.io/name=pmem-memcached -o jsonpath={..metadata.name})
+kubectl exec -n demo $pod grep 'I am PMEM.' /data/memcached-memory-file
```

Comment:
As I said before I think with these commands you can combine the output with the command as just put $ in front of the command.